Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -52,6 +52,41 @@ public class GraphTraverser: GraphTraversing {
         allTargets(excludingExternalTargets: true)
     }
 
+    public func filterIncludedTargets<GraphTargets: Collection>(
+        basedOn targets: GraphTargets,
+        testPlan: String?,
+        includedTargets: Set<String>,
+        excludedTargets: Set<String>,
```

Comment:
Is there a need to check for an intersection between `includedTargets` & `excludedTargets` so that we can throw an early error about misconfiguration?

Reply:
Let me know if you should we should still do that considering my comments below.

There could still be errors caught there for configurations that don't make sense, for example if `includedTargets` & `excludedTargets`'s intersection is non-`nil`, as it in this case the user would to include and exclude the same targets. Down to implement this check if we want it

Reply:
yep, I would implement it in the command

Reply:
@danyf90 I've added validation of that to the command